### DISCUSSION TITLE
Compare differences between two files

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#3988] [Ideas]
I have seen this mentioned elsewhere but there were no topics about this mentioned in the discussions section so I thought I would start one.

I am aware that there are separate applications for this but seems a little odd that there is not anything like this for Geany itself.  A way to compare the differences between two files is an extremely helpful tool but I have not really seen anything like this for this application.

Ideally it would be built in but I know this is something that would probably be offered more as a plugin because the ability to split Geany into two panes side by side (which would also be required for this compare tool) is also achieved by a plugin.

Closest thing I have seen is someone mentioned that there is a Lua script that then has to use a separate application to achieve this, which in this case would probably just use the application.
For this the chain of applications linked looks something like this for this current solution:
Geany - plugin manager - Lua support Plugin - Lua script to access compare tool - 3rd party compare tool - show results in Geany

Does anyone know of any simpler solutions for something like this?
Are there any plugins that can achieve this within one plugin?

Thank You

**Comment by elextr** [comment]
With the exception of a few plugins that are part of Geany most plugins are either part of the separate Geany-plugins collection repository or external plugins in their own repository owned by the plugin developer.  Discussions for plugins should be on the relevant repository, its most unlikely that a plugin will be added to Geany directly, they will be added to the Geany-plugins collection first, so thats where ideas and proposals (and PRs with code) for new plugins should happen.

AFAIK there is no diff plugin in Geany, the Geany-plugins collection or "well known" plugins outside it.

Closing since it should be discussed on a relevant plugin repository.

**Comment by ekotysh** [comment]
I agree with the poster, this should be a core feature of Geany and it's currently missing.
